healpy/src/_sphtools.pyx

```diff
@@ -301,6 +309,17 @@ def map2alm(m, lmax = None, mmax = None, niter = 3, use_weights = False,
     else:
         w_arr.allocAndFill(2 * nside, 1.)
 
+    cdef vector[double] full_weights
+    if pixel_weights_filename is not None:
+
+        full_weights = read_fullweights_from_fits(pixel_weights_filename.encode("UTF-8"), nside)
+        # pixel weighting requires 0 iterations
+        niter = 0
```

Comment:
@mreineck do you mean `niter` here?

Reply:
Most likely yes. For my tests I simply called map2alm() in sphtfunc.py with iter=0.